### ISSUE TITLE
Final Relationship item cannot be removed

### ISSUE THREAD

**Issue by morenoh149** [keystonejs/keystone#1074]
I have the following post model

``` javascript
var keystone = require('keystone'),
    Types = keystone.Field.Types,
    debug = require('debug')('r3dm:models:post');

var Post = new keystone.List('Post', {
  map: { name: 'title' },
  autokey: { path: 'slug', from: 'title', unique: true }
});

Post.add({
  cover: { type: Types.S3File },
  title: { type: String, required: true },
  state: { type: Types.Select, options: 'draft, published, archived', default: 'draft', index: true, required: true },
  author: { type: Types.Relationship, ref: 'User', index: true },
  publishedDate: { type: Types.Date, index: true, dependsOn: { state: 'published' } },
  content: {
    brief: { type: Types.Html, wysiwyg: true, height: 150 },
    extended: { type: Types.Html, wysiwyg: true, height: 400 }
  },
  translations: { type: Types.Relationship, ref: 'Post', many: true }
});

Post.schema.virtual('content.full').get(function() {
  return this.content.extended || this.content.brief;
});

Post.defaultSort = '-publishedDate';
Post.defaultColumns = 'title, state|20%, author|20%, publishedDate|20%';

Post.schema.pre('save', function(next) {
  var myPost = this,
      err;

  debug('saving Post:', myPost);
  if (myPost.state === 'published') {
    if (!myPost.publishedDate) {
      err = new Error('Cannot publish a post without a publishedDate.');
      next(err);
    }
    if (!myPost.author) {
      err = new Error('Cannot publish a post without an author.');
      next(err);
    }
  }
  next();
});

Post.register();
```

In my console I keep seeing this logged

```
  r3dm:models:post saving Post: +13s { _id: 54d1772e4e7f8ebe11d17b84,
  slug: 'aaa',
  title: 'aaa',
  __v: 0,
  translations: [ 54d1772e4e7f8ebe11d17b84 ],
  content:
   { brief: '<p>briefings</p>',
     extended: '<p>laksjf;laksdjf;laskjdf;alskjdfa</p>\r\n<p>&nbsp;</p>\r\n<p>&nbsp;</p>\r\n<p>asdklfajsdflaksaj</p>\r\n<p>extended</p>' },
  state: 'draft',
  cover: {} }
```

That's the model in the pre-save hook after removing all posts in the relationships box. Can anyone confirm this behavior? I'm running 0.2.x latest.
The post request is definitely sending an empty `translations` field. Interestingly as well is if you have 2 items in the array, you can delete one but you can't completely empty the array.
![screenshot 2015-02-17 14 14 30](https://cloud.githubusercontent.com/assets/587438/6238367/57f1d670-b6af-11e4-9b67-f956d84937a7.png)

**Comment by morenoh149** [comment]
This could be related to previous reports. Where it was debated what the return value of a form should be (paraphrasing) how `null` is distinct from `undefined`. I just observed a similar issue with a textarea. I can update (add/remove) content in the textarea. But I can't set the textarea to be completely empty.

**Comment by mxstbr** [comment]
#2644 also

**Comment by steevsachs** [comment]
I think this is caused by the use of `FormData` in `EditForm` and `UpdateForm`.  Null values are not recognized and so aren't submitted to the API for update.

I wonder if the correct approach might be to use the state of the component instead of the values of the DOM to identify and submit values.

**Comment by stennie** [comment]
Although this issue predates Keystone 4 development, a similar bug was introduced in 4.0.0-beta.5 and fixed in 4.0.0-rc.0 by way of #4332.

Regards,
Stennie
